### ISSUE TITLE
Volume meter value to decibel

### ISSUE THREAD

**Issue by Mayvis** [GoogleChromeLabs/web-audio-samples#257]
First, thanks for your sharing, I would like to know how to convert volume into decibel.
In your audioworklet volume meter example, I can get volume value.

https://github.com/GoogleChromeLabs/web-audio-samples/commit/7c109d7d21c24bfa031f843877837e36ea0028cb

```js
// volume need to * 100 to make sure value between 1 to 10
this._volume = Math.max(rms, this._volume * SMOOTHING_FACTOR) * 100;
 
// Is this the correct way to convert volume to decibel? My example is 16bits so I divided by 2^16 - 1
const dB = -10 * Math.log10(this._volume / 32767);  
```

**Comment by rvadhavk** [comment]
I don't think it's generally possible to compute the decibel level of microphone input without a reference value/calibration.  All you can really compute is the ratio of the current amplitude (or some running average) to max amplitude.

**Comment by Mayvis** [comment]
@rvadhavk yes, I want to measure the microphone input value. So I refer to this [link](https://dspillustrations.com/pages/posts/misc/decibel-conversion-factor-10-or-factor-20.html). 
In my case, _rms_ is the power not the _amplitude_. So I use below formula to measure it and luckily get good result.

![Screen Shot 2022-09-16 at 8 29 37 AM](https://user-images.githubusercontent.com/12667910/190531495-fc6f3d23-2a73-4d50-956d-1f8ae2ea3d6a.png)

I just curious, is this the correct way to get real or close dB value?

**Comment by rvadhavk** [comment]
> I just curious, is this the correct way to get real or close dB value? Or it's just ratio.
This isn't my area of expertise, so I could be wrong, but to get the "real" loudness in dB instead of just computing a ratio of current power to max power, I believe you need some reference measurement to know the relationship between the RMS of the actual sound pressure in pascals and the RMS of the unitless PCM data: https://en.wikipedia.org/wiki/Decibel#Acoustics.  So unless you can ask your users to play some sort of calibration sound at a known fixed distance from the microphone or you know exactly what microphone they're using and what the gain is and everything, you can't know the absolute loudness.

**Comment by kasperkamperman** [comment]
This won't be very reliable. For example with autoGainControl you already can influence this value. The output on MacOs Safari and MacOs Chrome is really different. 

I like the approach in this example (I started also with the volume-meter example) better and I assume it's even less resource intensive.

**Comment by Mayvis** [comment]
> This won't be very reliable. For example with autoGainControl you already can influence this value. The output on MacOs Safari and MacOs Chrome is really different. 
> 
> 
> 
> I like the approach in this example (I started also with the volume-meter example) better and I assume it's even less resource intensive. 

I prefer to use AudioWorklet instead of using autoGainControl😅. Because I also have to manipulate float32 to int16...etc. I worked on an AI company and build ASR associated project. I know the reliability that you mention. Thank you.
